### PR TITLE
Use LoRaWAN phy version is GS

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Use LoRaWAN phy version is GS. Covers the GS part of #4347 

#### Changes

- Get the phy based on the version in `TxRequest`.
- Update tests.

#### Testing

- UT
- Once we have the NS and/or PBA we should do some integration testing, esp with the AU/AS freq plans.

##### Regressions

Actual testing to ensure that downlinks work.

#### Notes for Reviewers

~⚠️ We shouldn't merge this without the NS/PBA parts.~

~Also I think that we should do this in `v3.14.1` since `v3.14.0` is already quite big.~

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/gatewayserver/io/io.go [hunk 1]
```diff
@@ -407,10 +407,17 @@ func (c *Connection) ScheduleDown(path *ttnpb.DownlinkPath, msg *ttnpb.DownlinkM
...
 		}
 	}
-	phy, err := band.GetByID(fp.BandID)
+
+	phyFromFP, err := band.GetByID(fp.BandID)
+	if err != nil {
+		return false, false, 0, err
+	}
+
+	phy, err := phyFromFP.Version(request.GetLorawanPhyVersion())
```
Comment (adriansmares) [inline pkg/gatewayserver/io/io.go]: I think this should fall back to `1.1-a` on `nil`, for backwards compatibility reasons.

**Review by adriansmares — COMMENTED** [review]
NS part has been added in https://github.com/TheThingsNetwork/lorawan-stack/pull/4349, which is merged.

File: pkg/gatewayserver/io/io.go (hunk 1 above)
Comment (KrishnaIyer) [reply pkg/gatewayserver/io/io.go]: I considered it. But I think that it's better to fail downlinks. If we just fallback and allow them, it'll be difficult to debug when customers come back with mismatched downlink issues. 
Also a `v3.14.x` GS is not required to be compatible with a `v3.13.x` NS right? 
What do you think @johanstokking?

Comment (johanstokking) [reply pkg/gatewayserver/io/io.go]: We should stay compatible with the old behavior, if it were only for Packet Broker and LoRaWAN Backend Interfaces support. This field can be optional, and if it isn't set, do not version the PHY and assume the latest version is used.

Comment (KrishnaIyer) [reply pkg/gatewayserver/io/io.go]: Ok sounds good. Changes coming up.

**Review by johanstokking — APPROVED** [review]

**Pull request merged**